src/QATools/Example/Pages/LoginPage.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * This file is part of the QA-Tools library.
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @copyright Michael Iwersen <mi.iwersen@gmail.com>
+ * @link      https://github.com/qa-tools/behat-example
+ */
+
+namespace QATools\Example\Pages;
+
+
+use QATools\QATools\HtmlElements\Element\Form;
+use QATools\QATools\HtmlElements\TypifiedPage;
+
+/**
+ * Class LoginPage.
+ *
+ * @page-url('/')
+ */
+class LoginPage extends TypifiedPage
+{
+
+	/**
+	 * The login form.
+	 *
+	 * @var Form
+	 * @find-by('id' => 'login-form')
+	 */
+	protected $loginForm;
+
+	/**
+	 * Fill the login form with given data.
+	 *
+	 * @param string $username The username.
+	 * @param string $password The password.
+	 *
+	 * @return static
+	 */
+	public function fillForm($username, $password)
+	{
+		$this->loginForm->fill(array(
+			'username' => $username,
+			'password' => $password,
+		));
+
+		return $this;
+	}
+
+	/**
+	 * Does the login.
+	 *
+	 * @return Page
+	 */
+	public function loginSuccess()
```

Comment:
> What is the third and fourth parameter about? I guess the last one is the to returned page.

See `login` method declaration in https://github.com/qa-tools/behat-example/pull/1/files#r24565821 comment.

> I guess this could be indirect part of the constraints we talked about , just as a manual check but how do we want to expose the the PageLocator or a similar function like resolvePage in the PageFactory. This is what we are exactly currently missing.

The `Page` has access to `PageFactory`. So the only thing we need to make visible from outside is `resolvePage` method.

> But how does your example differs to a successful login? and how to differentiate it in code?

Ha, you'll know once you see the `login` method definition: 3rd parameter is page to return on success and 4th parameter is page to return on failure.